### DISCUSSION TITLE
Constraint Formulas

### DISCUSSION THREAD

**Discussion by DeadlyDad** [hlorus/CAD_Sketcher#88] [Feature Requests]
(Note: I'm not set up to test if constraints can contain formulas or not.  If they are already, I can delete this.)

One issue that **MANY** CAD programs have is the inexplicable lack of formula evaluation in data entry boxes.  While there are plenty of times just putting in a number will do, there are plenty of times when having to use another app to calculate that number is annoying.  For example, if a hole needs to be made for a bolt with a shaft diameter of 102.50mm and a tolerance of 0.375mm, wouldn't it be easier to just type in `102.5 + 0.375`, rather than add it up in your head or on a calculator?  What's much more important than convenience is that just seeing '102.875' just tells you how big the hole is, **NOT** how big the bolt that goes through the hole should be or the required tolerance.  What if the bolt is changed to galvanized and now the shaft size changes to 102._6_?  Looking at the number might make you think that it will still fit because 102.6 **is** still smaller than 102.875.   ...but now the tolerance is below spec.  Not good.  By simply allowing the two variables to be separated in a formula, you make it obvious that increasing the shaft size by 0.1mm will require increasing the hole size as well, to maintain the required tolerance.  While adding two numbers together is a trivial example, the implications of specifying all of the variables that go into a dimension are decidedly **not** trivial.

...and, of course, having variable tables would expand this utility **exponentially.**  Being able to type `BoltM[8][ShaftDia] + BoltM[8][ShaftTolerance]` would not only make it **perfectly** clear what the hole is for, but changing it to a different sized bolt becomes trivial, especially if you change the '8' in the example to a variable that can then be altered in the table instead of chasing constraints.

**Comment by hlorus** [comment]
Blenders numeric input should be supported as described [here](https://docs.blender.org/manual/en/latest/scene_layout/object/editing/transform/control/numeric_input.html). I assume that's what you're asking for, right?
What's missing currently is drivers support which should allow to reference other dimension values

**Reply by DeadlyDad** [reply]
> Blenders numeric input should be supported as described [here](https://docs.blender.org/manual/en/latest/scene_layout/object/editing/transform/control/numeric_input.html). I assume that's what you're asking for, right? What's missing currently is drivers support which should allow to reference other dimension values

Having access to the full Python math library is great, but a GUI interface to create/edit dictionaries/lists/etc. that can be referenced in the formulas is also needed, as you've pointed out.

**Comment by hlorus** [comment]
Let's not try to reinvent the wheel here, it would go a long way if we could support blenders driver system.
